test/Driver/checkedc-notsupported.cpp

```diff
@@ -0,0 +1,9 @@
+// Checked C extension is not supported for C++.   Make sure driver
```

Comment:
I've added test cases forcing files to be compiled as different file types than the file extension specifies, with the Checked C extension flag enabled:

- I added a new test with a C file type. It is compiled as a regular C file and also as a C++ file..
- I added run lines where several file types are forced to be compiled as C files..
- I added run lines where the cl file type is compiled as several other language types.
- Finally, I also added tests for clang-cl testing the Microsoft command-line options /TP and /TC for compiling files as C++ and C files respectively.